plans/sep8.md

```diff
@@ -166,39 +164,34 @@ switch (approvalResponse.status) {
     showUser(approvalResponse);
     break;
   case ApprovalResponseStatus.action_required:
-    if (isBrowser) {
-      // To avoid popup blockers, the new window has to be opened directly in
-      // response to a user click event, so we need consumers to provide us a
-      // window instance that they created previously. This could also be done in
-      // an iframe or something.
-      const popup = window.open("", "name", "dimensions etc");
-
-      const actionResult = await approvalProvider.fetchActionInBrowser({
-        response: approvalResponse,
-        window: popup,
-      });
+    // Placeholder to show the message regarding why further actions are required.
+    showUser(approvalResponse.message);
 
-      showUser(actionResult);
+    // If the approval service requested specific fields, load the values of those fields without user intervention.
+    let actionFieldsValues = [];
+    if (!approvalResponse.action_fields || !approvalResponse.action_fields.length) {
+          actionFieldValues = approvalResponse.action_fields.map(field => loadValue(field););
+    }
 
-      // if action succeeded, submit transaction
-      if (actionResult.status === "success") {
-        transaction = new Transaction(actionResult.tx);
-        submitPayment(transaction);
+    if (!approvalResponse.action_method || approvalResponse.action_method === "GET") {
+      let actionUrl = approvalResponse.action_url;
+      if (!actionFieldsValues.length) {
+        actionUrl = buildUrlWithQueryParams(approvalResponse.action_url, approvalResponse.action_fields, actionFieldValues);
       }
-    } else if (isServerEnv || isNativeEnv) {
-      const actionRedirect = getActionUrl({
-        response: approvalResponse,
-        request: approvalRequest,
-        callbackUrl,
+
+      // The actual implementation regarding how to trigger the sending of the GET request and handle the response is up to the client.
+      sendGetRequestToActionUrl(actionUrl);
```

Comment:
❓ What is different about `sendGetRequestToActionUrl` at the end of the `GET` flow vs the `loadUrl` call at the end of the `POST` flow?

In both cases the client has a URL that it needs to take the user to and I thought they would be the same function. Is there other behavior that is unique to the `GET` case that happens here?

Reply:
> Is there other behavior that is unique to the GET case that happens here?

Nope. They are the same. Good catch. https://github.com/stellar/js-stellar-wallets/pull/201/commits/7e0acfc3d794433390425a797425e2b1e443fa91